src/main/java/io/charkchalk/backend/service/tag/TagController.java

```diff
@@ -0,0 +1,62 @@
+package io.charkchalk.backend.service.tag;
+
+import io.charkchalk.backend.entity.Tag;
+import io.charkchalk.backend.json.tag.BaseTagJson;
+import io.charkchalk.backend.json.tag.TagConverter;
+import io.charkchalk.backend.json.tag.TagJson;
+import io.charkchalk.backend.repository.TagRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.util.Optional;
+
+@Validated
+@RestController
+public class TagController {
+
+    @Autowired
+    private TagConverter tagConverter;
+
+    @Autowired
+    private TagRepository tagRepository;
+
+    @PostMapping("/api/tag")
+    public ResponseEntity<TagJson> createTag(@Valid @RequestBody BaseTagJson baseTagJson) {
+        Tag tag = tagConverter.convertToEntity(baseTagJson);
+        tagRepository.save(tag);
+        return ResponseEntity.ok(tagConverter.convertToJson(tag));
+    }
+
+    @GetMapping("/api/tag/{name}")
+    public ResponseEntity<TagJson> getTag(@PathVariable @NotNull String name) {
+        Optional<Tag> tagOptional = tagRepository.findByName(name);
+        return tagOptional.map(value -> ResponseEntity.ok(tagConverter.convertToJson(value)))
+                .orElseGet(() -> ResponseEntity.notFound().build());
+    }
+
+    @PutMapping("/api/tag")
```

Comment:
will it be better that match rules like other two route? cuz the action here is search by id, but others are search by name.